src/camel_case.rs

```diff
@@ -4,36 +4,34 @@
 ///  @returns {String} Returns the camel cased string.
 ///  @see lower_case, kebab_case, snake_case, start_case, upper_case, upper_first
 /// 
+extern crate regex;
 
+use camel_case::regex::Captures;
 
-// TODO: Rewrite just using regex
-pub fn new(string: String) -> String {
-    let mut combine_result: String = String::new();
-    let str_lower = string.to_ascii_lowercase();
-    let str_trim = str_lower.trim();
-    let mut str_array: Vec<char> = str_trim.chars().collect();
-
-    for i in 0..str_array.len() {
-        let char_to_string = str_array[i].to_string();
-        if char_to_string.trim().is_empty() {
-            // check the next character if number find next letter char
-            let mut index = i + 1;
-            while index <= str_array.len() {
-                if str_array[index].is_numeric() {
-                    index = index + 1;
-                    continue;
-                }
-                str_array[index] = str_array[index].to_ascii_uppercase();
-                break;
-            }
-            continue;
-        }
-        combine_result.push_str(&char_to_string)
+pub fn new(mut string: String) -> String{
+    //regex to substitute
+    let re = regex::Regex::new(r"(?x)
+        (?P<cut>[^a-zA-Z\d]+) # every char not in camelCase
+        (?:(?P<replace>
+            \d+[a-zA-Z]  # example: 24h
+            |
+            \d+    # example: 500
+            |
+            [a-zA-Z]  # example: format
+        )|$)").unwrap();
```

Comment:
awesome regex here! what formatter do you use? run `cargo fmt` to properly format

Reply:
ran through cargo fmt